test/integration/rmi_out_test.py

```diff
@@ -43,27 +51,146 @@ def test_deprish_out(rmi_out, request):
     deprish_to_ferc1 = rmi_out.grab_deprish_to_ferc1(clobber=True)
     assert not deprish_to_ferc1.empty
-    del deprish_to_ferc1
+
+
+def _add_data_source(df):
+    """
+    Add a data_source column with 'FERC' and 'PUC'.
+
+    The depreciation data includes a `data_source` column as a part of the
+    primary key of that data set. This is because there are sometimes we have
+    depreciation study from both `FERC` and `PUC` for the same utility and
+    plants. Because of this, in order to compare the FERC plant input tables to
+    the FERC-EIA-Depreciation tables we need to add *both* versions of the
+    depreciation `data_source`.
+    """
+    if 'data_source' not in df:
+        df = pd.concat([
+            df.assign(data_source='PUC'),
+            df.assign(data_source='FERC'),
+        ])
+    return df
+
+
+@pytest.mark.parametrize(
+    "df1_name,df2_name,data_col,by_name", [
+        (
+            "grab_deprish",
+            "grab_deprish_to_eia",
+            'plant_balance_w_common',
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_eia",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'utilities',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'plants',
+        ),
+    ]
+)
+def test_data_col_at_two_stages(
+    rmi_out,
+    df1_name: str,
+    df2_name: str,
+    data_col: str,
+    by_name: Literal['plants', 'utilities'],
+):
+    """
+    Test the consistency of a data column at two stages.
+
+    The data that is processed in this repo goes along multiple stages of its
+    journey. There are some quantities that are expected to be invariant when
+    aggregated at the plant level, and the utility level -- that the quantity
+    might slosh around between generators in a plant, or between plants within
+    a utility, but that when aggregated to the plant or utility level, it
+    should be the same at every step in the processing.
+
+    Ideally, the inputs
+    """
+    if by_name == 'plants':
+        by = PK_PLANTS
+    elif by_name == 'utilities':
+        by = PK_UTILS
+
```

Comment:
Need an `else` clause here that catches anything outside these two values and raises an error.